js/test/utils.test.js

```diff
@@ -132,103 +131,65 @@ describe('utils', () => {
+        assert.isTrue(arraysAreEqual);
+      }
     });
   });
 
   describe('#typedArraysAreEqual', () => {
     it('returns true if Uint8Arrays are equal', () => {
-      const arr1 = new Uint8Array([255, 255, 255]);
-      const arr2 = new Uint8Array([255, 255, 255]);
+      const { arr1, arr2 } = typedArraysAreEqual[0].input;
       const res = utils.typedArraysAreEqual(arr1, arr2);
       assert.isTrue(res);
     });
     it('returns false if Uint8Arrays are not equal', () => {
-      const arr1 = new Uint8Array([255, 255, 254]);
-      const arr2 = new Uint8Array([255, 255, 255]);
+      const { arr1, arr2 } = typedArraysAreEqual[1].input;
       const res = utils.typedArraysAreEqual(arr1, arr2);
       assert.isFalse(res);
     });
     it('throws error if any arrays are not of type Uint8Array', () => {
-      const arr1 = new Uint8Array([255, 255, 255]);
-      const arr2 = [255, 255, 255];
+      const { arr1 } = typedArraysAreEqual[2].input;
+      const arr2 = Array.from(typedArraysAreEqual[2].input.arr2);
       try {
         utils.typedArraysAreEqual(arr1, arr2);
         assert(false, 'expected an error');
       } catch (e) {
-        assert.include(e.message, 'Arrays must be of type Uint8Array');
+        assert.include(e.message, typedArraysAreEqual[2].errorMessage);
       }
     });
     it('returns false if Uint8Arrays lengths are not equal', () => {
-      const arr1 = new Uint8Array([255, 255]);
-      const arr2 = new Uint8Array([255, 255, 255]);
+      const { arr1, arr2 } = typedArraysAreEqual[3].input;
       const res = utils.typedArraysAreEqual(arr1, arr2);
       assert.isFalse(res);
     });
   });
 
   describe('#safeSlice', () => {
     it('returns a safe slice on an array', () => {
-      const arr = new Uint8Array([1, 2, 3, 4, 5]);
-      let res;
-      let arraysAreEqual;
-
-      // regular slice
-      res = utils.safeSlice(arr, 0, 3);
-      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([1, 2, 3]));
-      assert.isTrue(arraysAreEqual);
-
-      // slice that copies the original array
-      res = utils.safeSlice(arr);
-      arraysAreEqual = utils.typedArraysAreEqual(res, arr);
-      assert.isTrue(arraysAreEqual);
-
-      // slice with start index, but not end index
-      res = utils.safeSlice(arr, 2);
-      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([3, 4, 5]));
-      assert.isTrue(arraysAreEqual);
-    });
-    it('uses default values', () => {
-      const arr = new Uint8Array([1, 2, 3, 4, 5]);
-      let res;
-      let arraysAreEqual;
-
-      // default end
-      res = utils.safeSlice(arr, 3);
-      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([4, 5]));
-      assert.isTrue(arraysAreEqual);
-
-      // default start
-      res = utils.safeSlice(arr, null, 3);
-      arraysAreEqual = utils.typedArraysAreEqual(res, new Uint8Array([1, 2, 3]));
-      assert.isTrue(arraysAreEqual);
+      for (let i = 0; i < 5; i += 1) {
```

Comment:
Rather than hardcoding 5, can we do something like 

```
for (let i = 0; i < safeSlice.length; i += 1) {
    if (safeSlice[i].errorMessage) {
        try {} catch {}
    } else {
        normalTest
    }
```

Reply:
Same goes for other tests that are hardcoding indices

Reply:
I fixed this using `safeSliceError`, forgot to add that when I was moving over error data.